internal/server/data/provideruser_test.go

```diff
@@ -327,6 +329,158 @@ func TestListProviderUsers(t *testing.T) {
 				return provider.ID, &SCIMParameters{StartIndex: 1}, []models.ProviderUser{pu1, pu2}, 2
 			},
 		},
+		{
+			name: "equality filter",
+			setup: func(t *testing.T, tx *Transaction) (providerID uid.ID, p *SCIMParameters, expected []models.ProviderUser, totalCount int) {
+				provider := &models.Provider{
+					Name: "mockta",
+					Kind: models.ProviderKindOkta,
+				}
+
+				err := CreateProvider(tx, provider)
+				assert.NilError(t, err)
+
+				_ = createTestProviderUser(t, tx, provider, "david@example.com")
+				pu := createTestProviderUser(t, tx, provider, "lucy@example.com")
+				filter, err := filter.ParseFilter([]byte(fmt.Sprintf("id eq \"%d\"", pu.IdentityID)))
```

Comment:
Not a blocker, just an fyi, I always try to use tilde "raw string literals" when the string needs to include double quotes, to avoid having to escape them:

```go
fmt.Sprintf(`id eq "%d"`, pu.IdentityID)
```

Raw strings don't allow `\n` and those kinds of escapes, but otherwise work the same way.